File: packages/go-sdk/operations/api_keys.go
package operations

import "context"

// APIKeysService provides API key management operations.
type APIKeysService struct{ r Requester }

func NewAPIKeysService(r Requester) *APIKeysService { return &APIKeysService{r: r} }

func (s *APIKeysService) List(ctx context.Context, query map[string]string) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "GET", "/v1/api-keys", query, nil, nil, &result)
	return result, err
}

func (s *APIKeysService) Create(ctx context.Context, body any) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "POST", "/v1/api-keys", nil, nil, body, &result)
	return result, err
}

func (s *APIKeysService) Delete(ctx context.Context, keyID string) error {
	return s.r.DoRequest(ctx, "DELETE", PathParams("/v1/api-keys/{keyID}", map[string]string{"keyID": keyID}), nil, nil, nil, nil)
}

func (s *APIKeysService) Rotate(ctx context.Context, keyID string, body any) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "POST", PathParams("/v1/api-keys/{keyID}/rotate", map[string]string{"keyID": keyID}), nil, nil, body, &result)
	return result, err
}

File: packages/rust-sdk/src/operations/api_keys.rs
use crate::client::StraitClient;
use crate::errors::StraitError;
use crate::http::substitute_path_params;
use std::sync::Arc;

pub struct ApiKeysService {
    client: Arc<StraitClient>,
}

impl ApiKeysService {
    pub fn new(client: Arc<StraitClient>) -> Self {
        Self { client }
    }

    pub async fn list(&self, query: Option<&[(&str, &str)]>) -> Result<serde_json::Value, StraitError> {
        self.client.do_request("GET", "/v1/api-keys", query, None, None).await
    }

    pub async fn create(&self, body: serde_json::Value) -> Result<serde_json::Value, StraitError> {
        self.client.do_request("POST", "/v1/api-keys", None, None, Some(body)).await
    }

    pub async fn delete(&self, key_id: &str) -> Result<(), StraitError> {
        let path = substitute_path_params("/v1/api-keys/{keyID}", &[("keyID", key_id)]);
        self.client.do_request_no_content("DELETE", &path, None, None, None).await
    }

    pub async fn rotate(&self, key_id: &str, body: serde_json::Value) -> Result<serde_json::Value, StraitError> {
        let path = substitute_path_params("/v1/api-keys/{keyID}/rotate", &[("keyID", key_id)]);
        self.client.do_request("POST", &path, None, None, Some(body)).await
    }
}
